pkg/controller/daemon/controller.go

```diff
@@ -113,6 +114,9 @@ func NewDaemonSetsController(podInformer framework.SharedInformer, kubeClient cl
 	// TODO: remove the wrapper when every clients have moved to use the clientset.
 	eventBroadcaster.StartRecordingToSink(&unversionedcore.EventSinkImpl{Interface: kubeClient.Core().Events("")})
 
+	if kubeClient != nil && kubeClient.Core() != nil && kubeClient.Core().GetRESTClient() != nil {
```

Comment:
This would look a lot cleaner if `Core()`, `GetRESTClient()` and `GetRateLimiter()` handled a `nil` receiver and returned `nil`. Then you could just write:

```go
if rl := kubeClient.Core().GetRESTClient().GetRateLimiter(); rl != nil {
	metrics.RegisterMetricAndTrackRateLimiterUsage("daemon_controller", rl)
}
```